### PR TITLE
Update UI on remote survey changes

### LINKED ISSUE #2139: Survey doesn't reload when changed remotely
Changes to survey should be reflected everywhere:

Job cards
- [x] Job title
- [x] New ad hoc job
- [x] Removed ad hoc job
- [x] New normal job
- [x] Removed job with LOIs

Map
- [x] New LOIs
- [x] Deleted LOIs
- [x] LOI map feature default color <-- `Job` getting cached in `LocationOfInterest` makes this harder
- [x] LOI geometries (due to #2153)

Data collection flow

- [ ] Changed tasks (capture location -> drop pin)
- [ ] New tasks (add a question)

### PR DESCRIPTION
**Pull request by gino-m**
Towards #2139

Tested and working:

- [x] Items in survey list
- [x] Add job -> new LOI card
- [x] Remove  job -> job card gone
- [x] Add "add LOI" task -> new "add LOI" job card 
- [x] Remove "add LOI" task -> "add LOI" job card gone

PM guidance needed to present updates during data collection flow.

### CODE REVIEW

**Comment by gino-m** [comment]
Update: Code is working, tests need to be updated.

**Comment by gino-m** [comment]
Tests fixes. @shobhitagarwal1612 PTAL?

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt [hunk 1]
```diff
@@ -42,50 +43,59 @@ class SurveyRepositoryTest : BaseHiltTest() {
...
     surveyRepository.selectedSurveyId = SURVEY.id
     advanceUntilIdle()
 
     surveyRepository.activeSurveyFlow.test { assertThat(expectMostRecentItem()).isEqualTo(SURVEY) }
+    assertThat(surveyRepository.activeSurvey).isEqualTo(SURVEY)
   }
 
-  @Test
-  fun activeSurveyFlow_emitsEmptyOnClearActiveSurvey() = runWithTestDispatcher {
+  fun `clearActiveSurvey() resets active survey`() = runWithTestDispatcher {
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt:64]: Please add the `@Test` annotation here.

**Review by shobhitagarwal1612 — APPROVED** [review]
One small comment. Rest LGTM

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt:64]: Good catch!

**Pull request merged**